File: Program2/HeapNode.cs
using System;

namespace ConsoleApp1
{
    class HeapNode
    {
        private int m_id; //process id
        private int m_priority; // process priority
        private int m_timems; // process run time ms

        /*
         * Constructor
         */
         public HeapNode(int id, int priority, int timems)
        {
            m_id = id;
            m_priority = priority;
            m_timems = timems;
        }

        /*
         * Get id
         */
         public int GetId()
        {
            return m_id;
        }

        /*
         *  Get priority
         */
         public int GetPriority()
        {
            return m_priority;
        }

        /*
         * Get Time
         */
         public int GetTimems()
        {
            return m_timems;
        }

        /*
         * Override string for a more sensible print
         */
        public override string ToString()
        {
            return "HeapNode: " + m_id + " " + m_priority + " " + m_timems;
        }
    }
}

File: Program1/HeapNode.java
/*
 * A simple heap node class to simulate a process
 * contains an id, priority and sleep time to simulate runtime
 */
public class HeapNode {
    private int m_id; // process id
    private int m_priority; // process priority
    private int m_timems; // run time

    /*
     * Constructor
     */
    public HeapNode(int id, int priority, int timems) {
        m_id = id;
        m_priority = priority;
        m_timems = timems;
    }

    /*
     * get method for id
     */
    public int getId() {
        return m_id;
    }

    /*
     * get method for priority
     */
    public int getPriority() {
        return m_priority;
    }

    /*
     * get method for runtime
     */
    public int getTime() {
        return m_timems;
    }

    /*
     * override of to string to do a more sensible print statement
     */
    @Override
    public String toString() {
        String print = String.format("Process(id=%d, priority=%d, runtime=%d)", m_id, m_priority, m_timems);
        return print;
    }
}
